solr/core/src/test/org/apache/solr/handler/TestSQLHandler.java

```diff
@@ -1853,21 +1853,18 @@ public void testParallelTimeSeriesGrouping() throws Exception {
   }
 
   protected List<Tuple> getTuples(final SolrParams params, String baseUrl) throws IOException {
-    //log.info("Tuples from params: {}", params);
-    TupleStream tupleStream = new SolrStream(baseUrl, params);
-
-    tupleStream.open();
-    List<Tuple> tuples = new ArrayList<>();
-    for (; ; ) {
-      Tuple t = tupleStream.read();
-      //log.info(" ... {}", t.fields);
-      if (t.EOF) {
-        break;
-      } else {
-        tuples.add(t);
+    List<Tuple> tuples = new LinkedList<>();
+    try (TupleStream tupleStream = new SolrStream(baseUrl, params)) {
```

Comment:
This pattern makes me wish we had an `asStream` method on TupleStream that returned a java Stream. Maybe a future issue.

Reply:
Maybe? Here I'm fixing a bug where the stream didn't get closed if an exception gets raised, causing tests to hang.